packages/snaps-execution-environments/src/common/lockdown/lockdown-events.ts

```diff
@@ -0,0 +1,45 @@
+// When creating a sandbox, limitation of the events from leaking
+// sensitive objects is required. This is done by overriding own properties
+// of prototypes of all existing events.
+
+import { logError } from '@metamask/snaps-utils';
+
+/**
+ * Targeted Event objects and properties.
+ * Note: This is a map of the prototypes that inherit from Events with
+ * properties that are identified to leak sensitive objects.
+ */
+export const targetEvents: { [index: string]: any } = Object.freeze({
+  UIEvent: ['view'],
+  MutationEvent: ['relatedNode'],
+  MessageEvent: ['source'],
+  FocusEvent: ['relatedTarget'],
+  MouseEvent: ['relatedTarget', 'fromElement', 'toElement'],
+  TouchEvent: ['targetTouches', 'touches'],
+  Event: ['target', 'currentTarget', 'srcElement', 'composedPath'],
+});
+
+/**
+ * Attenuate Event objects by replacing its own properties.
+ */
+export function executeLockdownEvents() {
```

Comment:
There are couple of reasons why I added it that way initially.

So, the first try/catch block wraps everything and I agree that it should re-throw and I added it.

The nested try/catch block (the second one) I added because some of the events specified for the security are not available in some web browsers which we're supporting. For example, I found that `TouchEvent` is not available in Firefox, so it will crash. So, I changed it now to an `if` condition checking whether or not the given Event object is available in `globaThis`, then taking an action for securing it.

I do want for the events to be secured, but I also **don't** want for Snaps architecture to crash in some web browsers because of missing Event

Let me know what you think about this approach I added now or any other solutions that might be better for the case.

Reply:
Given that I added `if` that will check the existence of the Event object, I couldn't find any case in which this would throw an error. So, I removed it since I cannot cover it with tests (no working scenario which throws an error). Feel free to reopen this question if you think this should be different.

Reply:
yea as long as you verify the proto actually exists before trying to override it you shouldn't have a problem 👍